### ISSUE TITLE
[Improvement] output current project reference in `conan info`

### ISSUE THREAD

**Issue by azriel91** [conan-io/conan#109]
The conan file reference `project/0.1.0@user/channel` is annoying to type, as there isn't a way to autocomplete each segment in the string. I find myself often having to type the ref for a project I am working on.

For projects that use `conanfile.py`, a sensible reference string is deducible from the conan file if they specify the project name and a version, but for projects that use `conanfile.txt`, there cannot be a project name or version, so there is not be enough information to calculate a sensible reference string -- you may get something like `None/None@azriel91/testing` for example.

Currently when typing `conan info [path/to/project]`, the conan file reference is not output for the conanfile inside the project directory. I propose changing this command to:

* output the conan file reference and dependency information when a `conanfile.py` is present in the project directory that contains *both* a name and a version:

    ```
    $ conan info googlemock-conan/
    googlemock/1.7.0@azriel91/testing
        URL: https://github.com/azriel91/googlemock-conan.git
        Required by:
        Requires:
            googletest/1.7.0@azriel91/stable-1
    googletest/1.7.0@azriel91/stable-1
        URL: https://github.com/google/googletest-conan.git
        Required by:
            Project
    ```

* not changing the output for projects that use `conanfile.txt`, or projects with `conanfile.py` that do not specify a name or version

**Comment by azriel91** [comment]
Current pull request implements what I've written, but up for debate is whether or not that is the right thing to do.

**Comment by memsharded** [comment]
Released in 0.7, tested and everything ok.
